### PR TITLE
Check global subscriptions's readiness on ready method.

### PR DESCRIPTION
**Pull request by delgermurun**
Check this out. If there is anything to improve, please tell me.

### CODE REVIEW

File: client/router.js [hunk 1]
```diff
@@ -173,29 +173,25 @@ Router.prototype.middleware = function(middlewareFn) {
...
 
   var subscriptions;
   if(arguments.length === 0) {
-    subscriptions = _.values(currentRoute.getAllSubscriptions());
+    subscriptions = _.values(globalRoute.getAllSubscriptions());
+    subscriptions.concat(_.values(currentRoute.getAllSubscriptions()));
   } else {
     subscriptions = _.map(arguments, function(subName) {
-      return currentRoute.getSubscription(subName);
+      return globalRoute.getSubscription(subName) || currentRoute.getSubscription(subName);
```
Comment (arunoda) [inline client/router.js]: I think can check this is a better way. Like this:

~~~js
function checkSubs(subs) {
  return _.every(subs, function(sub) {
    return sub.ready();
  });
}

var isReady = 
  checkSubs(currentRoute.getAllSubscriptions) &&
  checkSubs(globalRoute.getAllSubscriptions);

return isReady;
~~~

I think that's simple.
What do you think?

File: client/router.js [hunk 2]
```diff
@@ -173,29 +173,25 @@ Router.prototype.middleware = function(middlewareFn) {
...
-    if(!sub) {
-      return false;
-    } else if(sub.ready() === false) {
-      return false;
-    }
-  }
-
-  return true;
+  return _.every(subscriptions, function(sub) {
+    return sub && sub.ready();
```
Comment (arunoda) [inline client/router.js]: No need to check for sub.

Comment (delgermurun) [inline client/router.js]: How about wrong name? So sub is `undefined`.

Comment (arunoda) [inline client/router.js]: Hmm. good one.

**Comment by arunoda** [comment]
Okay. We are good to go.

**Comment by arunoda** [comment]
Did a new release: v1.0.1 with this fix.

**Comment by delgermurun** [comment]
:+1:

**Pull request merged**